crates/solver/src/settlement.rs

```diff
@@ -469,6 +545,102 @@ pub mod tests {
         }
     }
 
+    #[test]
+    pub fn satisfies_price_checks_sorts_out_invalid_prices() {
```

Comment:
Maybe a unit test to check some of the other paths like:
- Order where one token is included but not the other
- Order where where no tokens are included